app/js/lib/server/index.js

```diff
@@ -0,0 +1,106 @@
+/* global URLSearchParams */
+
+'use strict';
+
+import EventDispatcher from '../common/event-dispatcher';
+
+import Settings from './settings';
+import Network from './network';
+import WebPush from './webpush';
+import API from './api';
+import Reminders from './reminders';
+
+// Private members.
+const p = Object.freeze({
+  // Private properties.
+  settings: Symbol('settings'),
+  net: Symbol('net'),
+  webPush: Symbol('webPush'),
+  api: Symbol('api'),
+});
+
+export default class Server extends EventDispatcher {
+  constructor({ settings, net } = {}) {
+    super(['online']);
+
+    // Private properties.
+    this[p.settings] = settings || new Settings();
+    this[p.net] = net || new Network(this[p.settings]);
+    this[p.api] = new API(this[p.net], this[p.settings]);
+    this[p.webPush] = new WebPush(this[p.api], this[p.settings]);
+
+    // Init
+    this.reminders = new Reminders(this[p.api], this[p.settings]);
+
+    this[p.net].on('online', (online) => this.emit('online', online));
+    this[p.webPush].on('message', (msg) => this.emit('push-message', msg));
+
+    window.server = this;
```

Comment:
We do, we replace some components code by their minified counterpart in production (React in particular does many unnecessary checks useful for dev but not in prod) and the code is not minified in dev. See in `gulpfile.js` the `build-dev` and `build-production` tasks.